zips/draft-pacu-regtest-local-consensus-definition.md

```diff
@@ -0,0 +1,313 @@
+```
+  ZIP: Unassigned
+  Title: Regtest: Definition of a Local Consensus test mode for Zcash Full-Nodes
+  Owners: ZIP Editors
+  Original-Authors: Francisco Gindre <pacu@zecdev.org>
+  Credits: Daira-Emma Hopwood
+           Zancas (Zingo Labs)
+           idky137 (Zingo Labs)
+           Kris Nuttycombe
+  Status: Draft
+  Category: Consensus
+  Created: 2025-02-26
+  License: MIT
+  Pull-Request: <https://github.com/zcash/zips/pull/986>
+```
+# Regtest: Definition of a Local Consensus test mode for Zcash Full-Nodes
+
+
+## Terminology 
+
+The key words "MUST", "REQUIRED", "MUST NOT", "SHOULD", and "MAY" in this document
+are to be interpreted as described in BCP 14 [^BCP14] when, and only when, they 
+appear in all capitals.
+
+The terms "Mainnet" and "Testnet" in this document are to be interpreted as defined 
+in the Zcash protocol specification [^protocol-networks].
+
+## Abstract
+Regtest is a testing mode inherited from the Bitcoin node. It provides a way to 
+have a consensus node on a local network with a private state that a developer 
+can control in order to reproduce certain situations deterministically for 
+testing purposes. Regtest, as it was implemented on Zcashd, is similar to a 
+“testnet” but without miners and remote peers.
+
+## Motivation
+It is necessary to define Regtest mode so that different implementations of Zcash
+Full Nodes provide the same capabilities so that testing infrastructure can be 
+interoperable. Regtest mode was inherited from Bitcoin functionality and never 
+defined in a ZIP due to resource constraints and priorities. Differences between 
+Zebra and Zcashd Regtest implementations found by developers involved in Zcashd 
+deprecation motivated this ZIP as a way to document existing assumptions, 
+expectations and functionalites of Regtest, and to document regtest-only features.
+This ZIP documents differences in behavior, values, expectations and assumpions 
+in comparison to Testnet. It uses the Zcash protocol and existing ZIPs as guiding 
+principles for the document structure. It also documents pre-existing and 
+previously *undocumented* requirements, constants and values that are present in
+the Zcashd 6.1.0 implementation and `zcash_protocol` crate 0.4.3[^zcash_protocol].
+This ZIP should be useful for Core Developers working on full node implementations
+but also a handbook for developers and QA testers to make use and take advantage
+of Regtest functionality to ensure test coverage of their codebases and a high 
+quality assurance to Zcash users.
+
+
+## Requirements
+1. Nodes launched in Regtest mode MUST be able to generate deterministic sequences
+of blocks and transactions. 
```

Comment:
How about:

"Finalizers, launched in regtest mode, MUST be able to **efficiently** generate deterministic sequences of valid blocks of transactions."